tests/test_check_dagger.py

```
@@ -0,0 +1,327 @@
+
+
+def blockinfile(filename: str | pathlib.Path, contents: str, *, prefix: str = None, suffix: str = None):
+    """This is similar to the function in
+     * https://homely.readthedocs.io/en/latest/ref/files.html#homely-files-blockinfile-1
+     * ansible.modules.lineinfile
+
+    Not used now, but it will be useful if we want to generate "test_" function for each notebook.
+    """
+    begin = end = -1
+
+    lines = open(filename, "rt").readlines()
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)